module/tools/RoboCupConfiguration/src/RoboCupConfiguration.cpp

```diff
@@ -0,0 +1,506 @@
+        hostname = utility::support::get_hostname();
+
+        // Wifi interface
+        wifi_interface = utility::support::get_wireless_interface();
+
+        // IP Address
+        ip_address = utility::support::get_ip_address(wifi_interface);
+
+        // Team ID
+        YAML::Node global_config = YAML::LoadFile(get_config_file("GlobalConfig.yaml"));
+        team_id                  = global_config["team_id"].as<int>();
+        player_id                = global_config["player_id"].as<int>();
+
+        // SSID
+        ssid = utility::support::get_ssid(wifi_interface);
+
+        // Password
+        password = utility::support::get_wifi_password(ssid, wifi_interface);
+
+        // Robot position
+        {
+            std::string soccer_file = get_config_file("Soccer.yaml");
+            YAML::Node config       = YAML::LoadFile(soccer_file);
+            robot_position          = config["position"].as<std::string>();
+        }
+        // Ready position
+        {
+            std::string ready_file = get_config_file(robot_position.get_config_name());
+            YAML::Node config      = YAML::LoadFile(ready_file);
+            ready_position         = config["ready_position"].as<Eigen::Vector3d>();
+        }
+
+        // Check if we have permissions
+        if (geteuid() != 0) {
+            log_message = "Warning: To configure the network, run with sudo.";
+        }
+    }
+
+    void RoboCupConfiguration::configure_network() {
+        // In case values aren't set yet, set them
+        set_config_values();
+
+        // Check if we are on a robot
+        if (get_platform() != "nugus") {
+            log_message = "Configure Error: Network configuration only available on NUgus robots!";
+            return;
+        }
+
+        // Check if we have permissions
+        if (geteuid() != 0) {
+            log_message = "Configure Error: Insufficient permissions! Run with sudo!";
+            return;
+        }
+
+        // Copy the network files
+        std::filesystem::copy_file(fmt::format("system/{}/etc/systemd/network/30-wifi.network", hostname),
+                                   "/etc/systemd/network/30-wifi.network",
+                                   std::filesystem::copy_options::overwrite_existing);
+        std::filesystem::copy_file(
+            fmt::format("system/{}/etc/wpa_supplicant/wpa_supplicant-{}.conf", hostname, wifi_interface),
+            fmt::format("/etc/wpa_supplicant/wpa_supplicant-{}.conf", wifi_interface),
+            std::filesystem::copy_options::overwrite_existing);
+
+        // Restart the network
+        system("systemctl restart systemd-networkd");
+        system("systemctl restart wpa_supplicant");
+        system(("systemctl restart wpa_supplicant@" + wifi_interface).c_str());
+
+        log_message = "Network configured!";
+    }
+
+    void RoboCupConfiguration::set_config_values() {
+        /* GAME CONFIG */
+        {  // Write the robot's position to the soccer file
```

Comment:
Why scope?

Reply:
so the config and file variables are contained and don't exist outside